FWCore/Framework/interface/GlobalSchedule.h

```diff
@@ -119,6 +120,9 @@ namespace edm {
     std::vector<edm::propagate_const<WorkerPtr>> extraWorkers_;
     ProcessContext const* processContext_;
     unsigned int numberOfConcurrentLumis_;
+    unsigned int numberOfConcurrentRuns_;
+    unsigned int numberOfConcurrentProcessBlocks_ = 1;
+    unsigned int numberOfConcurrentJobs_ = 1;
```

Comment:
All the global WorkerManagers are in one vector. These 4 numberOf* variables are only used to size that vector and calculate the index to the correct WorkerManager in the vector when one is needed.  It is designed this way partially because it extends the existing pattern and also it is convenient because some of the operations on the WorkerManagers involve a simple loop over them all and then doing the same thing to each.

I thought defining the 4 variables in a parallel way would make it easier to understand, but I guess it is confusing because the last two are always 1. Would making the last 2 const be enough? Or I should I just make them actual constexpr constants in the GlobalSchedule.cc file and think up better names for them.

Reply:
I'd make the`numberOfConcurrentProcessBlocks_` and `numberOfConcurrentJobs_` `static constexpr` constants here, and include a comment along "we have no intention to have concurrent ProcessBlocks or Jobs, but having these constants following the existing naming convention is convenient when operating on WorkerManagers". 

Of course if you can come up with better names for them, that would be great, but I think these are acceptable (when explained with a comment).

Reply:
Done. I slightly reworded the comment.